expression/builtin_time.go

```diff
@@ -2773,6 +2773,9 @@ func (b *builtinExtractDurationSig) evalInt(row chunk.Row) (int64, bool, error)
 		return 0, isNull, err
 	}
 	res, err := types.ExtractDurationNum(&dur, unit)
+	if dur.IsNeg() {
```

Comment:
I think maybe fixing the sign issue in function `types.ExtractDurationNum` is closer to the "root cause".